pkg/metrics/metrics_collector_test.go

```diff
@@ -0,0 +1,65 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package metrics
+
+import (
+	"testing"
+	"time"
+
+	"gotest.tools/assert"
+
+	"github.com/apache/incubator-yunikorn-core/pkg/metrics/history"
+)
+
+func TestHistoricalPartitionInfoUpdater(t *testing.T) {
+	metricsHistory := history.NewInternalMetricsHistory(3)
+	setInternalMetricsCollectorTickerForTest(500 * time.Millisecond)
+	metricsCollector := NewInternalMetricsCollector(metricsHistory)
+	metricsCollector.StartService()
+
+	go func() {
+		metrics := GetSchedulerMetrics()
+		i := 0
+		for i < 3 {
+			metrics.IncTotalApplicationsRunning()
+			metrics.AddAllocatedContainers(2)
+			i += 1
+			time.Sleep(400 * time.Millisecond)
```

Comment:
Do we really need to sleep between each update? Extends the test run unneeded

Reply:
I wanted to wait for at least 2 cycles. Do you think it is still an issue after ticker has been configured to 10msec (from comment below)?

Reply:
with a shorter sleep I am OK with the test and the wait we put in.